### PR TITLE
Remove some clusters from data-model/silabs and add data-model/chip

### PR DESCRIPTION
**Pull request by vivien-apple**
#### Problem

This is a split of #7873 that contains the removals of various clusters as well as new types from the spec. But the current types are not changed. I will focus on the types change in #7873

#### Change overview
 * Add `src/app/zap-templates/zcl/data-model/chip` and move the zap definitions from `src/app/zap-templates/zcl` there
 * Remove some definitions from `src/app/zap-templates/zcl/data-model/Silabs`
 * Move types definition from `src/app/zap-templates/zcl/data-model/Silabs/types.xml` to `src/app/zap-templates/zcl/data-model/chip/chip-types.xml` and add some extra types from the spec
 * Update gen/ code

#### Testing
This was tested manually by running `./scripts/tests/test_suite.sh`. This is mostly code removal so I won't expect any side effects if it builds.

### CODE REVIEW

**Comment by vivien-apple** [comment]
The changes in `src/app/util/basic-types.h` are only re-ordering alphabetically the types and the addition of new types. I have done it in this PR in order to make it easier to spot which types will be updated in #7873

File: src/app/zap-templates/zcl/data-model/chip/chip-types.xml [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    <type id="0x29" description="Signed 16-bit integer"    name="int16s"            size="2"  analog="true"   />
+    <type id="0x2A" description="Signed 24-bit integer"    name="int24s"            size="3"  analog="true"   />
+    <type id="0x2B" description="Signed 32-bit integer"    name="int32s"            size="4"  analog="true"   />
+    <type id="0x2C" description="Signed 40-bit integer"    name="int40s"            size="5"  analog="true"   />
+    <type id="0x2D" description="Signed 48-bit integer"    name="int48s"            size="6"  analog="true"   />
+    <type id="0x2E" description="Signed 56-bit integer"    name="int56s"            size="7"  analog="true"   />
+    <type id="0x2F" description="Signed 64-bit integer"    name="int64s"            size="8"  analog="true"   />
+    <type id="0x30" description="8-bit enumeration"        name="enum8"             size="1"  discrete="true" />
+    <type id="0x31" description="16-bit enumeration"       name="enum16"            size="2"  discrete="true" />
+    <type id="0x38" description="Semi-precision"           name="float_semi"        size="2"  analog="true"   />
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/chip-types.xml:56]: This semi-precision type doesn't exist in Matter, fwiw.

File: src/app/zap-templates/zcl/data-model/chip/general-diagnostics-cluster.xml [hunk 2]
```diff
@@ -53,6 +53,23 @@ limitations under the License.
...
     <item name="SoftwareReset" value="0x06"/>
   </enum>  
+  <enum name="InterfaceType" type="ENUM8">
+    <item name="Unspecified" value="0x00"/>
+    <item name="WiFi" value="0x01"/>
+    <item name="Ethernet" value="0x02"/>
+    <item name="Cellular" value="0x03"/>
+    <item name="Thread" value="0x04"/>
+  </enum>
+  <struct name="NetworkInterfaceType">
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/general-diagnostics-cluster.xml:63]: This one is weird; in the spec it's just "NetworkInterface", so not sure why "Type" is added here.  Presumably this is just copy/paste, but maybe a followup to fix that?

**Review by bzbarsky-apple — APPROVED** [review]

**Comment by bzbarsky-apple** [comment]
@msandstedt @andy31415 @mspang @Damian-Nordic @jmartinez-silabs Could you please take a look?

File: src/app/zap-templates/zcl/data-model/chip/chip-types.xml (hunk 1 above)
Comment (vivien-apple) [reply src/app/zap-templates/zcl/data-model/chip/chip-types.xml:56]: > This semi-precision type doesn't exist in Matter, fwiw.

I have tried to keep all the types from ZCL at the moment. There are many types that does not exists in Matter and will be removed in #7873

File: src/app/zap-templates/zcl/data-model/chip/general-diagnostics-cluster.xml (hunk 2 above)
Comment (vivien-apple) [reply src/app/zap-templates/zcl/data-model/chip/general-diagnostics-cluster.xml:63]: Will do a followup. I guess the name can be updated either when `CHAR_STRING` is implemented for struct or when `HWADR` type is implemented.

**Review by jmartinez-silabs — APPROVED** [review]

**Comment by woody-apple** [comment]
@saurabhst ?

**Review by woody-apple — APPROVED** [review]

**Review by saurabhst — APPROVED** [review]

**Pull request merged**